catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/query/QueryApplication.java

```diff
@@ -109,6 +133,73 @@ public void init() {
           res.header("Content-Encoding", "gzip");
         });
 
+    post(
+        "/cql/transform/:transformerId",
+        (req, res) -> {
+          String transformerId = req.params(":transformerId");
+          String body = util.safeGetBody(req);
+          CqlRequest cqlRequest = mapper.readValue(body, CqlRequest.class);
+          CqlQueryResponse cqlQueryResponse = executeCqlQuery(cqlRequest);
+
+          for (ServiceReference<QueryResponseTransformer> queryResponseTransformer :
```

Comment:
❗️ This is actually pretty confusing. At first blush I thought we were iterating over all `QueryResponseTransofrmer`s and collecting the transformed data into a response. Though it appears that only the first match is used since we actually return from inside the `for` loop. Which also means that the code after the loop doesn't get executed if there's a matching transformer? Is this correct?

Reply:
@leo-sakh i could try that but right now im trying out joe's comment that's using `bundleContext.getServiceReferences(QueryResponseTransformer.class.getName(), "(id=<transformerId>)")` not sure which is better

Reply:
Might be able to use both...if it makes sense to do so